full/src/main/java/apoc/cypher/CypherExtended.java

```diff
@@ -246,40 +246,46 @@ private void runDataStatementsInTx(
             BlockingQueue<RowResult> queue,
             Map<String, Object> params,
             boolean addStatistics,
-            long timeout,
             boolean reportError,
             String fileName) {
         while (scanner.hasNext()) {
             String stmt = removeShellControlCommands(scanner.next());
             if (stmt.trim().isEmpty()) continue;
-            boolean schemaOperation;
-            try {
-                schemaOperation = isSchemaOperation(stmt);
-            } catch (Exception e) {
-                collectError(queue, reportError, e, fileName);
-                return;
-            }
 
-            if (!schemaOperation) {
-                if (isPeriodicOperation(stmt)) {
-                    Util.inThread(pools, () -> {
-                        try {
-                            return db.executeTransactionally(
-                                    stmt, params, result -> consumeResult(result, queue, addStatistics, tx, fileName));
-                        } catch (Exception e) {
-                            collectError(queue, reportError, e, fileName);
-                            return null;
-                        }
-                    });
-                } else {
+            // Periodic operations cannot be schema operations, so no need to check that here (will fail as invalid
+            // query)
+            if (isPeriodicOperation(stmt)) {
+                Util.inThread(pools, () -> {
+                    try {
+                        return db.executeTransactionally(
+                                stmt, params, result -> consumeResult(result, queue, addStatistics, tx, fileName));
+                    } catch (Exception e) {
+                        collectError(queue, reportError, e, fileName);
+                        return null;
+                    }
+                });
+            } else {
+                AtomicBoolean isSchemaError = new AtomicBoolean(false);
+                try {
                     Util.inTx(db, pools, threadTx -> {
                         try (Result result = threadTx.execute(stmt, params)) {
                             return consumeResult(result, queue, addStatistics, tx, fileName);
                         } catch (Exception e) {
-                            collectError(queue, reportError, e, fileName);
+                            // APOC historically skips schema operations
+                            if (!e.getMessage().contains("Schema operations on database 'neo4j' are not allowed")) {
```

Comment:
`on database 'neo4j'`, what about databases that has another name?

Reply:
Ah yes 🙈